n3fit/src/n3fit/vpinterface.py

```diff
@@ -45,9 +46,111 @@
     "T35",
 ]
 
+
 class N3Stats(MCStats):
+    """The PDFs from n3fit are MC PDFs
+    however, since there is no grid, the CV has to be computed manually"""
+
+    def error_members(self):
+        return self.data
+
     def central_value(self):
-        return self.data[0]
+        return np.mean(self.data, axis=0)
+
+
+class N3LHAPDFSet(LHAPDFSet):
+    """Extension of LHAPDFSet using n3fit models"""
+
+    def __init__(self, name, pdf_models):
+        log.info("Creating LHAPDF-like n3fit PDF")
+        self._error_type = "replicas"
+        self._name = name
+        self._lhapdf_set = pdf_models
+        self._flavors = None
+        self.basis = check_basis("evolution", EVOL_LIST)["basis"]
+
+    def xfxQ(self, x, Q, n, fl):
+        """Return the value of the PDF member for the given value in x"""
+        return self.grid_values([fl], [x]).squeeze()[n]
```

Comment:
The use of `n` for the replica is quite common: https://github.com/NNPDF/nnpdf/blob/2ae49281c5d471be490ed5c7bbd28e9211091c19/libnnpdf/src/lhapdfset.cc#L85 (and then later in vp) and since this should basically substitute a vp PDF it's better to keep the same name. Also because not be a replica but a member of some hessian pdf for instance, it just happens to be always a replica in this subclass.

Reply:
Okay, not ideal but fair enough